src/ArrayManifold.jl

```diff
@@ -76,6 +84,14 @@ function inner(M::ArrayManifold, x, v, w; kwargs...)
     return inner(M.manifold, array_value(x), array_value(v), array_value(w))
 end
 
+function exp(M::ArrayManifold, x, v; kwargs...)
+    is_manifold_point(M, x; kwargs...)
+    is_tangent_vector(M, x, v; kwargs...)
+    y = ArrayMPoint(exp(M.manifold, array_value(x), array_value(v)))
+    is_manifold_point(M, y; kwargs...)
+    return y
+end
+
```

Comment:
This looks quite nice. The only question I have is: Do we want to check every time? I think we discussed a while back, that _with_ types we do checks, without types (so if you have just arrays) we don't. Can that be done, here, too?

Reply:
I don't think I understand? This method of `exp` looks just like the other function for array manifolds.

Reply:
Yes, but array manifolds was intended to explicitly turn on checks (i.e. loosing a little performance compared to just arrays). Here you always have the checks on. Should we do something about that here and not check things?

Reply:
I think it would be better to discuss this in a separate issue, it shouldn't block this PR.